cloud/aws/deploy/up.go

```diff
@@ -237,6 +221,35 @@ func (d *DeployServer) Up(request *deploy.DeployUpRequest, stream deploy.DeployS
 		}
 		principals[v1.ResourceType_Function] = execPrincipals
 
+		// Deploy all buckets
+		buckets := map[string]*bucket.S3Bucket{}
+		for _, res := range request.Spec.Resources {
+			switch b := res.Config.(type) {
+			case *deploy.Resource_Bucket:
+				buckets[res.Name], err = bucket.NewS3Bucket(ctx, res.Name, &bucket.S3BucketArgs{
+					// TODO: Calculate stack ID
+					StackID: stackID,
+					Bucket:  b.Bucket,
+				})
+				if err != nil {
+					return err
+				}
+
+				if len(b.Bucket.Notifications) > 0 {
+					_, err = bucket.NewS3Notification(ctx, fmt.Sprintf("notification-%s", res.Name), &bucket.S3NotificationArgs{
+						StackID:      stackID,
+						Location:     details.Region,
+						Bucket:       buckets[res.Name],
+						Functions:    execs,
+						Notification: b.Bucket.Notifications,
+					})
+					if err != nil {
+						return err
+					}
+				}
```

Comment:
Is is simpler to deploy all notifications at once? Did we want to separate these out by target like you have for GCP

Reply:
Pulumi only supports deploying one S3 Notification per bucket.